core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -131,19 +132,37 @@ SubscriptionImpl subscribe(SubscriptionImpl subscription) {
 
     @Override
     public void abort() {
+        abort0(AbortedStreamException::get);
+    }
+
+    @Override
+    public void abort(Throwable cause) {
+        requireNonNull(cause, "cause");
+        abort(() -> cause);
+    }
+
+    @Override
+    public void abort(Supplier<? extends Throwable> causeSupplier) {
+        requireNonNull(causeSupplier, "causeSupplier");
+        abort0(causeSupplier);
+    }
+
+    private void abort0(Supplier<? extends Throwable> causeSupplier) {
+        final Throwable cause = requireNonNull(causeSupplier.get(),
```

Comment:
What is the point of using `Supplier` if you’re going to call `get()` regardless of whether `abort()` will fail?

Reply:
`causeSupplier.get()` is deferred until creating `new CloseEvent(cause)`.